packages/admin-frontend/lib/fetch.js

```diff
@@ -7,11 +7,113 @@
  */
 
 import UniversalCookie from 'universal-cookie';
+import fetch from 'isomorphic-unfetch';
+import type { BookDetails, Category, RemoteData, Chapter } from '../types';
+import { bookApiUrl } from '../config';
 
 const JWT_KEY = 'jwt';
-
 const Cookie = () => new UniversalCookie();
 
 export function getTokenFromLocalCookie() {
   return Cookie().get(JWT_KEY);
 }
+
+export async function fetchBook(
+  id: string | number,
+  language: string
+): Promise<RemoteData<BookDetails>> {
+  return await doFetch(`${bookApiUrl}/books/${language}/${id}`);
+}
+
+// Because the backend model and business logic for categories doesn't play nice together
+const bookCategoryMapper = book => {
+  const category: Category = book.categories.find(
+    c => c.name === 'classroom_books'
+  )
+    ? 'classroom_books'
+    : 'library_books';
+
+  book.category = category;
+  return book;
+};
+
+/*
+* Wrap fetch with some error handling and automatic json parsing
+*/
+async function doFetch(
+  url: string,
+  options: ?{
+    method: 'POST' | 'GET' | 'PUT',
+    body: ?any
+  }
+): Promise<RemoteData<any>> {
+  const token = process.browser ? getTokenFromLocalCookie() : undefined;
+
+  const response = await fetch(url, {
+    headers: {
+      Authorization: token ? `Bearer ${token}` : null
+    },
+    ...options
+  });
+
+  let result;
+  if (response.headers.get('Content-Type').includes('application/json')) {
+    result = await response.json();
+  } else {
+    result = await response.text();
+  }
+
+  if (response.ok) {
+    return {
+      data: result,
+      isOk: true,
+      statusCode: response.status
+    };
+  }
+
+  return {
+    error: result,
+    isOk: false,
+    statusCode: response.status
+  };
+}
+
+export async function saveBook(
```

Comment:
Since we moved editing to this frontend, the old save method isn't in use anymore in gdl-frontend and should be removed there.